### ISSUE TITLE
Feature Request: Scroll completion candidates with mouse wheel

### ISSUE THREAD

**Issue by rswgnu** [company-mode/company-mode#742]
If the mouse pointer is within the company completion popup, then it would be great if the mouse wheel scrolled through candidates.

**Comment by dgutov** [comment]
Would be nice to have, I agree.

**Comment by DivineDominion** [comment]
I tried a couple of variations of this:

    (define-key company-active-map (kbd "<wheel-up>") #'scroll-up)

Doesn't matter which key map I use, scrolling doesn't quite work.

Changing the selection does, though:

    (define-key company-active-map (kbd "<wheel-up>") #'company-select-next)
    (define-key company-active-map (kbd "<wheel-down>") #'company-select-previous)

Depending on your scroll preference, you might need to reverse the directions :)

My testing showed that `company-posframe` requires:

    (define-key company-posframe-active-map (kbd "<wheel-up>") #'company-select-next)
    (define-key company-posframe-active-map (kbd "<wheel-down>") #'company-select-previous)

**Comment by dgutov** [comment]
@DivineDominion

> Depending on your scroll preference, you might need to reverse the directions :)

Yeah, the other direction seems more natural to me. I also wonder whether scrolling should move faster (by 3?), or scroll by pages like e.g. `company-next-page`.

Another thing that seems missing here is the check regarding where the cursor is, like in `company-select-mouse`. So a couple of dedicated commands seem to be necessary anyway.

> My testing showed that company-posframe prefers to scroll buffer contents instead of the selection:
> % This equivalent binding does not work!

They do seem to work, but only when the mouse is outside of the frame. Perhaps posframe's author could help. Though child frames' behavior is finicky at times and not stable across desktop environments/OSes.

**Comment by dgutov** [comment]
The new [child frame](https://github.com/company-mode/company-mode/pull/1525) frontend supports scrolling with the mouse.
